src/Analyzers/CSharp/Analyzers/RemoveUnreachableCode/RemoveUnreachableCodeHelpers.cs

```diff
@@ -12,15 +14,35 @@ internal static class RemoveUnreachableCodeHelpers
     {
         public static ImmutableArray<ImmutableArray<StatementSyntax>> GetSubsequentUnreachableSections(StatementSyntax firstUnreachableStatement)
         {
-            SyntaxList<StatementSyntax> siblingStatements;
+            StatementSyntax[] siblingStatements;
             switch (firstUnreachableStatement.Parent)
             {
                 case BlockSyntax block:
-                    siblingStatements = block.Statements;
+                    siblingStatements = block.Statements.ToArray();
                     break;
 
                 case SwitchSectionSyntax switchSection:
-                    siblingStatements = switchSection.Statements;
+                    siblingStatements = switchSection.Statements.ToArray();
+                    break;
+
+                case GlobalStatementSyntax globalStatement:
+                    if (globalStatement.Parent is not CompilationUnitSyntax compilationUnit)
+                    {
+                        return ImmutableArray<ImmutableArray<StatementSyntax>>.Empty;
+                    }
+
+                    var builder = ArrayBuilder<StatementSyntax>.GetInstance();
+                    foreach (var member in compilationUnit.Members)
+                    {
+                        if (member is not GlobalStatementSyntax currentGlobalStatement)
+                        {
+                            continue;
+                        }
```

Comment:
Note to self: I think it would be good to have a test capturing the difference between `continue;` and `break;`. That's an error scenario where there are global statements, followed by a class declaration, followed by global statements.

Another test case (error scenario too) is having a class declaration as the first member, followed by global statements.